capsules/src/i2c_master.rs

```diff
@@ -4,19 +4,28 @@ use enum_primitive::enum_from_primitive;
 use kernel::common::cells::{MapCell, OptionalCell, TakeCell};
 use kernel::hil::i2c;
 use kernel::{
-    CommandReturn, Driver, ErrorCode, Grant, ProcessId, Read, ReadWrite, ReadWriteAppSlice, Upcall,
+    CommandReturn, Driver, ErrorCode, Grant, GrantDefault, ProcessId, ProcessUpcallFactory, Read,
+    ReadWrite, ReadWriteAppSlice, Upcall,
 };
 
 /// Syscall driver number.
 use crate::driver;
 pub const DRIVER_NUM: usize = driver::NUM::I2cMaster as usize;
 
-#[derive(Default)]
 pub struct App {
     callback: Upcall,
     slice: ReadWriteAppSlice,
 }
 
+impl GrantDefault for App {
+    fn grant_default(_process_id: ProcessId, cb_factory: &mut ProcessUpcallFactory) -> Self {
+        App {
+            callback: cb_factory.build_upcall(1).unwrap(),
```

Comment:
Same here?

Reply:
See other comment, this value is correct